### PR TITLE
feat: Add JSON schema to `gcs` destination plugin

### PR DESCRIPTION
**Pull request by candiduslynx**
Closes https://github.com/cloudquery/cloudquery/issues/16407

Basically, an adapted version of https://github.com/cloudquery/cloudquery/pull/16404

### CODE REVIEW

**Review by erezrokah — APPROVED** [review]

**Review by disq — CHANGES_REQUESTED** [review]
main.go not updated with WithJSONSchema?

**Comment by candiduslynx** [comment]
@disq added schema to plugin init

**Review by disq — APPROVED** [review]

File: plugins/destination/gcs/client/spec/schema_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+			Name: "csv file spec",
+			Spec: `{
+	"format": "csv",
+	"format_spec": {
+		"skip_header": true,
+		"delimiter": "#"
+	},
+	"path": "abc",
+    "bucket": "abc"
+}`,
```
Comment (disq) [inline plugins/destination/gcs/client/spec/schema_test.go:23]: erm, indent?

Comment (candiduslynx) [reply plugins/destination/gcs/client/spec/schema_test.go:23]: it doesn't matter here + we use the raw string

**Pull request merged**